### PR TITLE
Provide valid C# example and remove C++/CLI function definition.

### LINKED ISSUE #1120: "Implementing the GetInstance Method" on ICustomMarshaler needs language-specific example
This sample code snippet currently uses C++ for all languages (C#, VB), so copy and paste does not work.

https://github.com/dotnet/docs/blob/master/xml/System.Runtime.InteropServices/ICustomMarshaler.xml#L117

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
Fixes #1120 

/cc @dotnet/docs @elinor-fung @jkoritzinsky

### CODE REVIEW

**Review by jkoritzinsky — APPROVED** [review]

**Comment by AaronRobinsonMSFT** [comment]
@gewarren PTAL. Thanks!

File: xml/System.Runtime.InteropServices/ICustomMarshaler.xml [hunk 1]
```diff
@@ -124,11 +124,9 @@ const IID IID_IOld = {0x9B2BAADD,0x0705,0x11D3,{0xA0,0xCD,0x00,0xC0,0x4F,0xA3,0x
...
+ In addition to implementing the <xref:System.Runtime.InteropServices.ICustomMarshaler> interface, custom marshalers must implement a `static` method called `GetInstance` that accepts a <xref:System.String> as a parameter and has a return type of <xref:System.Runtime.InteropServices.ICustomMarshaler>. This `static` method is called by the common language runtime's COM interop layer to instantiate an instance of the custom marshaler. The string that is passed to `GetInstance` is a cookie that the method can use to customize the returned custom marshaler. A minimal, but complete, <xref:System.Runtime.InteropServices.ICustomMarshaler> implementation in the form of the `NewOldMarshaler` type is below.
```
Comment (gewarren) [inline xml/System.Runtime.InteropServices/ICustomMarshaler.xml:127]:
```suggestion
 In addition to implementing the <xref:System.Runtime.InteropServices.ICustomMarshaler> interface, custom marshalers must implement a `static` method called `GetInstance` that accepts a <xref:System.String> as a parameter and has a return type of <xref:System.Runtime.InteropServices.ICustomMarshaler>. This `static` method is called by the common language runtime's COM interop layer to instantiate an instance of the custom marshaler. The string that is passed to `GetInstance` is a cookie that the method can use to customize the returned custom marshaler. The following example shows a minimal, but complete, <xref:System.Runtime.InteropServices.ICustomMarshaler> implementation. 
```

**Review by gewarren — APPROVED** [review]
Looks good, just tweaked the intro to the example.

File: xml/System.Runtime.InteropServices/ICustomMarshaler.xml (hunk 1 above)
Comment (AaronRobinsonMSFT) [reply xml/System.Runtime.InteropServices/ICustomMarshaler.xml]: Much better. Thank you!

**Pull request merged**